modules/storage/local.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2020 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package storage
+
+import (
+	"io"
+	"net/url"
+	"os"
+	"path/filepath"
+)
+
+var (
+	_ ObjectStorage = &LocalStorage{}
+)
+
+// LocalStorage represents a local files storage
+type LocalStorage struct {
+	dir string
+}
+
+// NewLocalStorage returns a local files
+func NewLocalStorage(bucket string) (*LocalStorage, error) {
+	if err := os.MkdirAll(bucket, os.ModePerm); err != nil {
+		return nil, err
+	}
+
+	return &LocalStorage{
+		dir: bucket,
+	}, nil
+}
+
+// Open a file
+func (l *LocalStorage) Open(path string) (io.ReadCloser, error) {
+	f, err := os.Open(filepath.Join(l.dir, path))
+	if err != nil {
+		return nil, err
+	}
+	return f, nil
```

Comment:
I'm fairly certain you can just do:

```suggestion
	return os.Open(filepath.Join(l.dir, path))
```